File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/paymentproduct/validation/Validator.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.paymentproduct.validation;

import java.io.Serializable;

/**
 * POJO which holds the Validator data.
 * Containing all the validation types.
 */
public class Validator implements Serializable {

	private static final long serialVersionUID = 8524174888810141991L;

	private ExpirationDate expirationDate;
	private EmailAddress emailAddress;
	private IBAN iban;
	private FixedList fixedList;
	private Length length;
	private Luhn luhn;
	private Range range;
	private RegularExpression regularExpression;
	private TermsAndConditions termsAndConditions;

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public Validator() {}

	public ExpirationDate getExpirationDate(){
		return expirationDate;
	}

	public EmailAddress getEmailAddress(){
		return emailAddress;
	}

	public FixedList getFixedList(){
		return fixedList;
	}

	public IBAN getIBAN(){ return iban; }

	public Length getLength(){
		return length;
	}

	public Luhn getLuhn(){
		return luhn;
	}

	public Range getRange(){
		return range;
	}

	public RegularExpression getRegularExpression(){
		return regularExpression;
	}

	public TermsAndConditions getTermsAndConditions() {
		return termsAndConditions;
	}

}

File: sync/Worldline/repo/onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/paymentproduct/validation/Validator.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.paymentproduct.validation

import java.io.Serializable

/**
 * Data class which holds the Validator data.
 * Containing all the validation types.
 */
data class Validator(
    val expirationDate: ExpirationDate? = null,
    val emailAddress: EmailAddress? = null,
    val iban: IBAN? = null,
    val fixedList: FixedList? = null,
    val length: Length? = null,
    val luhn: Luhn? = null,
    val range: Range? = null,
    val regularExpression: RegularExpression? = null,
    val termsAndConditions: TermsAndConditions? = null
) : Serializable {

    companion object {
        @Suppress("Unused")
        private const val serialVersionUID = 8524174888810141991L
    }
}
